node/overseer/src/lib.rs

```diff
@@ -884,14 +889,20 @@ where
 	fn handle_external_request(&mut self, request: ExternalRequest) {
 		match request {
 			ExternalRequest::WaitForActivation { hash, response_channel } => {
-				// We use known leaves here because the `WaitForActivation` message
-				// is primarily concerned about leaves which subsystems have simply
-				// not been made aware of yet. Anything in the known leaves set,
-				// even if stale, has been activated in the past.
-				if self.known_leaves.peek(&hash).is_some() {
+				if self.active_leaves.get(&hash).is_some() {
```

Comment:
shouldn't we look at both ? The initial comment makes sense, otherwise we might wait for an activation which already happened ?

Reply:
known_leaves should be a superset of active

Reply:
known leaves just exists so we can detect a chain reversion (where a previously active leaf becomes active again).

If we rely on known leaves, versus active leaves - then we don't actually wait until the leaf becomes active again (which was the idea), but instead forward the create_inherent call immediately.